### PR TITLE
apps/wm_test: Add command to test all internal functions of wifi_manager

### PR DESCRIPTION
**Pull request by SeonghoByeon**
- Add all event-driven state transitions inside WiFi Manager
- Add scanning callback registration module for initialization of softap mode
- Remove wifi_net.c/h (WiFi does not need to handle ipv4 addr)
- Add MAC address type modification tool in WiFi Manager (string <-> byte)
- Fetch rssi from the device driver properly in WiFi Manager

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [327a2ba0370076c93ee68a156517ab449c3bcfc1] -

**Comment by seinfra** [comment]
Target : [327a2ba0370076c93ee68a156517ab449c3bcfc1] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [327a2ba0370076c93ee68a156517ab449c3bcfc1] - Code Rule Check Result:
apps/examples/wifi_manager_sample/wm_test.c:675: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)

**Comment by seinfra** [comment]
Target : [3c41039368518e3a64e6dffd2f15a882401b98ca] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3c41039368518e3a64e6dffd2f15a882401b98ca] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3c41039368518e3a64e6dffd2f15a882401b98ca] -

**Comment by seinfra** [comment]
Target : [3be1581f5d1f9312f3422fe6def741a6cf59c85f] -

**Comment by seinfra** [comment]
Target : [3be1581f5d1f9312f3422fe6def741a6cf59c85f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3be1581f5d1f9312f3422fe6def741a6cf59c85f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b1a2a62960e5fbe9c60ec540fcadf6a7c6ae67d6] -

**Comment by seinfra** [comment]
Target : [b1a2a62960e5fbe9c60ec540fcadf6a7c6ae67d6] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b1a2a62960e5fbe9c60ec540fcadf6a7c6ae67d6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [5215cda5705219c378f520d97b899f23b4eab40d] -

**Comment by seinfra** [comment]
Target : [5215cda5705219c378f520d97b899f23b4eab40d] -

**Comment by seinfra** [comment]
Target : [cff094492e95b052412a6ff07173133861a927b4] -

**Comment by seinfra** [comment]
Target : [cff094492e95b052412a6ff07173133861a927b4] -

**Comment by seinfra** [comment]
Target : [cff094492e95b052412a6ff07173133861a927b4] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [cff094492e95b052412a6ff07173133861a927b4] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cff094492e95b052412a6ff07173133861a927b4] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cff094492e95b052412a6ff07173133861a927b4] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b8f0e1417d349b4d93d3e5f2f196cdc5f034f3b9] -

**Comment by seinfra** [comment]
Target : [38bb9f5f8dbd7b4395b748570edbcfa91c9f205e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [38bb9f5f8dbd7b4395b748570edbcfa91c9f205e] -

**Comment by seinfra** [comment]
Target : [36bb61bae817e91fbe29d2c539d9ba9b6d786433] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [36bb61bae817e91fbe29d2c539d9ba9b6d786433] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [36bb61bae817e91fbe29d2c539d9ba9b6d786433] -

File: framework/include/wifi_manager/wifi_manager.h [hunk 1]
```diff
@@ -290,6 +290,34 @@ wifi_manager_result_e wifi_manager_get_config(wifi_manager_ap_config_s *config);
  */
 wifi_manager_result_e wifi_manager_remove_config(void);
 
+/**
+ * @brief convert mac address (48bit) to mac address string (FF:FF:FF:FF:FF:FF)
+ *
+ * @param[in]  mac_addr  :  mac address 48bit
```
Comment (sunghan-chang) [inline framework/include/wifi_manager/wifi_manager.h]: You might need to check this syntax on doxygen.
Doxygen get first param and second one separately and show them on different tabs.
```:``` can cause strange format on doxygen.

File: framework/src/st_things/things_stack/cloud/cloud_connector.c [hunk 2]
```diff
@@ -56,33 +56,38 @@ int CICheckDomain(const char *DomainName, char **pIP)
...
+	} 
 
-		usleep(500);
-		if (pIP != NULL) {
-			*pIP = things_strdup(ipbuffer);
-		}
+	THINGS_LOG_E(TAG, " Failed to get the IP, hard coding the ip");
+	snprintf(ipbuffer, sizeof(ipbuffer), "%s", "13.124.51.231");
+
+	usleep(500);
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/cloud/cloud_connector.c]: To use ```usleep``` function, you should include <unistd.h>

File: framework/src/st_things/things_stack/cloud/cloud_connector.c [hunk 3]
```diff
@@ -56,33 +56,38 @@ int CICheckDomain(const char *DomainName, char **pIP)
...
+	if (pIP != NULL) {
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/cloud/cloud_connector.c]: for NULL, <sys/types.h>

File: framework/src/st_things/things_stack/cloud/cloud_connector.c (hunk 2 above)
Comment (SeonghoByeon) [reply framework/src/st_things/things_stack/cloud/cloud_connector.c]: Hi, @YiHyunJin.
Could you check the comment above?

File: framework/src/st_things/things_stack/cloud/cloud_connector.c (hunk 3 above)
Comment (SeonghoByeon) [reply framework/src/st_things/things_stack/cloud/cloud_connector.c]: Hi, @YiHyunJin.
Could you check the comment above, related with NULL?

**Comment by seinfra** [comment]
Target : [3b8527b12bc71f9861a2ce2140d17e81e23ca882] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3b8527b12bc71f9861a2ce2140d17e81e23ca882] - Code Rule Check Result:
framework/src/st_things/things_stack/cloud/cloud_connector.c:60: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)

File: framework/src/st_things/things_stack/cloud/cloud_connector.c (hunk 2 above)
Comment (SeonghoByeon) [reply framework/src/st_things/things_stack/cloud/cloud_connector.c]: @sunghan-chang @YiHyunJin <unistd.h> inserted above.

File: framework/src/st_things/things_stack/cloud/cloud_connector.c (hunk 3 above)
Comment (SeonghoByeon) [reply framework/src/st_things/things_stack/cloud/cloud_connector.c]: @sunghan-chang @YiHyunJin <sys/types.h> inserted above.

**Comment by seinfra** [comment]
Target : [689eb6695456076d12591ecb1f31b168a9ba865f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [689eb6695456076d12591ecb1f31b168a9ba865f] - Code Rule Check OK.

File: framework/include/wifi_manager/wifi_manager.h (hunk 1 above)
Comment (KimDongEon) [reply framework/include/wifi_manager/wifi_manager.h]: I have confirmed that: is normally displayed in doxygen.

ex)
Parameters
[in]   mac_addr : mac address 48bit
[out] mac_str    : mac address string

**Pull request merged**